Shared/Samples/Show device location with NMEA data sources/ShowDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,296 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import ExternalAccessory
+
+extension ShowDeviceLocationWithNMEADataSourcesView {
+    /// The model used to store the geo model and other expensive objects used in this view.
+    class Model: ObservableObject {
```

Comment:
consider making the Model MainActor

Reply:
I tried to add `@MainActor` to the data model type, but encountered several "Call to main actor-isolated instance method in a synchronous non-isolated context" problem. I think it is due to the fact that the several detached tasks are interacting with the data source. If the data model is main actor, then the detached tasks cross boundary and make it unsafe.

Currently, I applied `@MainActor` to properties and methods that need to be on the main thread.

If we all agree, I can also refactor the code to not use detached tasks so I can mark the entire data model class `@MainActor`.

Mark intended to try out detached tasks from the discussion this thread: https://github.com/Esri/arcgis-maps-sdk-swift-samples/issues/153#issuecomment-1503802682 but I can change that.

Reply:
I've made the model main actor.